packages/react/src/components/sideNavigation/SideNavigation.test.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { axe } from 'jest-axe';
+
+import { SideNavigation } from './SideNavigation';
+import { IconHome } from '../../icons';
+
+const labels = {
+  mainLevel1: 'Main level 1',
+  mainLevel2: 'Main level 2',
+  mainLevel3: 'Main level 3',
+  subLevel1: 'Sub level 1',
+  subLevel2: 'Sub level 2',
+  toggleButton: 'Navigate to page',
+};
+const url = '#';
+
+const renderSideNavigation = () =>
+  render(
+    <SideNavigation id="sideNavigation" toggleButtonLabel={labels.toggleButton}>
+      <SideNavigation.MainLevel id="mainLevel1" icon={<IconHome />} label={labels.mainLevel1}>
+        <SideNavigation.SubLevel id="subLevel1" active href={url} label={labels.subLevel1} />
+      </SideNavigation.MainLevel>
+      <SideNavigation.MainLevel
+        id="mainLevel2"
+        href={url}
+        icon={<IconHome />}
+        label={labels.mainLevel2}
+        external
+        openInExternalDomainAriaLabel="Url points to external website"
+      />
+    </SideNavigation>,
+  );
+
+const queryButton = (key: string) => screen.queryByRole('button', { name: labels[key] });
+const queryLink = (key: string) => screen.queryByRole('link', { name: labels[key] });
+
+describe('<SideNavigation /> spec', () => {
+  it('renders the component', () => {
+    const { asFragment } = renderSideNavigation();
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should not have basic accessibility issues', async () => {
+    const { container } = renderSideNavigation();
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
+
+  test('should open main level if sub level is active', () => {
```

Comment:
Previous tests apply the _it_ instead of _test_. Im not too familiar with their differences, is there a reason why we wish to use _test_ here and onwards?

Reply:
Probably copy-paste thingy:
- [ ] use consistent test commands throughout tests